components/802.15.4_RF/atmel-rf-driver/source/NanostackRfPhyAtmel.cpp

```diff
@@ -321,6 +321,13 @@ __asm static void delay_loop(uint32_t count)
   BCS  %BT1
   BX   lr
 }
+#elif defined (__ARMCC_VERSION) /* ARMC6 */
+void delay_loop(uint32_t count)
```

Comment:
In which world this doesn't get optimized out ([ref](https://godbolt.org/z/nl3tfn)) ?

Reply:
It does, I just needed a placeholder to get it building with ARMC6.

Other compilers already have assembly version provided, but none of those builded OK with ARMC6